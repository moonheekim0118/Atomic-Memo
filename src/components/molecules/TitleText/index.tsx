import React from 'react';
import useValidation from '../../../hooks/useValidation';
import styled from 'styled-components';
import Textarea from '../../atoms/Textarea';
import Span from '../../atoms/Span';

const TitleText = () => {
  const [value, onChange, error] = useValidation({ max: 40, min: 1 });

  return (
    <StyledTitleText>
      <Textarea value={value} onChange={onChange} size="title" />
      <Count>
        <Span color={error ? 'red' : 'black'}>{`${value.length}/40`}</Span>
      </Count>
    </StyledTitleText>
  );
};

const StyledTitleText = styled.div`
  position: relative;
  display: flex;
  justify-content: space-between;
  border-bottom: 2px solid #f4f4f4;
`;

const Count = styled.div`
  position: absolute;
  top: -23px;
  left: 10px;
`;

export default TitleText;
