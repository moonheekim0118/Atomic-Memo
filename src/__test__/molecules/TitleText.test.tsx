import 'jsdom-global/register';
import React from 'react';
import TitleText from '../../components/molecules/TitleText';
import { mount } from 'enzyme';

describe('<TitleText/>', () => {
  let container = null;
  const mockFn = jest.fn();

  it('renders correctly', () => {
    container = mount(
      <TitleText value="Work" onChange={mockFn} error={false} />
    );
  });

  it('matches snpashot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
