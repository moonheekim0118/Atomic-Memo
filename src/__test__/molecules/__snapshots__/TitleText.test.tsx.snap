// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TitleText/> matches snpashot 1`] = `"<div class=\\"TitleText__StyledTitleText-sc-5531il-0 bvunhV\\"><textarea rows=\\"2\\" font-size=\\"2rem\\" placeholder=\\"Title\\" class=\\"Textarea__StyledTextArea-o90l14-0 duiURD\\">Work</textarea><div class=\\"TitleText__Count-sc-5531il-1 hyqDxi\\"><span color=\\"black\\" width=\\"auto\\" display=\\"inline\\" class=\\"Span__StyledSpan-sc-10y5m6u-0 jaanRG normal\\">4/40</span></div></div>"`;
